Eatery/Controllers/Eateries/Campus/ScrollableViewController.swift

```diff
@@ -0,0 +1,374 @@
+        tabBar.backgroundColor = .white
+        tabBar.layer.zPosition = 2
+
+        tabBar.snp.makeConstraints { make in
+            make.height.equalTo(40)
+        }
+
+        tabBarStack = UIStackView(arrangedSubviews: [headerContainer, tabBar])
+        tabBarStack.axis = .vertical
+    }
+
+    private func createMenuStack() {
+        menuItemsView = UIView()
+        menuItemsView.backgroundColor = .white
+
+        menuItemsStack = UIStackView(arrangedSubviews: originalViews)
+        menuItemsStack.axis = .vertical
+        menuItemsView.addSubview(menuItemsStack)
+        view.addSubview(menuItemsView)
+        view.addSubview(tabBarStack) // Tabbar added here to account for z position
+
+        tabBarStack.snp.makeConstraints { make in
+            make.top.equalToSuperview()
+            make.leading.trailing.equalToSuperview()
+        }
+
+        menuItemsView.snp.makeConstraints { make in
+            make.top.equalTo(tabBar.snp.bottom)
+            make.leading.trailing.bottom.equalToSuperview()
+        }
+
+        menuItemsStack.snp.makeConstraints { make in
+            make.top.bottom.equalToSuperview()
+            make.leading.trailing.equalToSuperview().inset(5)
+        }
+
+    }
+
+    private func createHighToLowStack() {
+        let reversedItems = Array(allItems.reversed())
+        let sortedVC = CampusEateryExpandedMenuViewController(eatery: eatery, category: "", menu: reversedItems)
+
+        menuItemsHighToLowStack = UIStackView(arrangedSubviews: [sortedVC.view])
+        menuItemsHighToLowStack.axis = .vertical
+        menuItemsView.addSubview(menuItemsHighToLowStack)
+        menuItemsHighToLowStack.isHidden = true
+
+        self.menuItemsHighToLowStack.snp.makeConstraints { make in
+            make.edges.equalToSuperview()
+        }
+    }
+
+    private func createLowToHighStack() {
+        let sortedVC = CampusEateryExpandedMenuViewController(eatery: eatery, category: "", menu: allItems)
+
+        menuItemsLowToHighStack = UIStackView(arrangedSubviews: [sortedVC.view])
+        menuItemsLowToHighStack.axis = .vertical
+        menuItemsView.addSubview(menuItemsLowToHighStack)
+        menuItemsLowToHighStack.isHidden = true
+
+        self.menuItemsLowToHighStack.snp.makeConstraints { make in
+            make.edges.equalToSuperview()
+        }
+    }
+
+    @objc func manualScroll() {
+        let currentPos = self.scrollView?.contentOffset.y
+        let viewControllerPosOnParent = self.view.frame.origin.y
+        let viewPos = originalViews[tabBar.selectedSegmentIndex].frame.minY
+        let minOffset: CGFloat = 4
+        let targetPos = viewPos - self.scrollOffset + viewControllerPosOnParent + minOffset
+        let topOfSelf = viewControllerPosOnParent - self.scrollOffset
+        let pixPS: CGFloat = 2500
+
+        self.isManualScrolling = true
+
+        if self.scrollView?.contentOffset.y ?? 0 > topOfSelf {
```

Comment:
you already stored this value in currentPos, use that constant instead

also, instead of using the nil coalescing operator (??) everytime you access currentPos, just use it once when setting currentPos equal to the scrollView's vertical content offset.